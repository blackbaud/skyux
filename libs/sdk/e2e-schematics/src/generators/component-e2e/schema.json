{
  "$schema": "http://json-schema.org/schema",
  "$id": "library-e2e",
  "title": "Create a library e2e",
  "description": "Creates an e2e infrastructure for a library.",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the library.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What is the name of the library?",
      "pattern": "^[a-zA-Z].*$"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the library (used for linting)."
    }
  },
  "additionalProperties": false,
  "required": ["name"]
}
