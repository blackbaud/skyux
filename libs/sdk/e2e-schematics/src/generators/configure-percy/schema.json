{
  "$schema": "http://json-schema.org/schema",
  "$id": "configure-percy",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "e2e project name",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    }
  }
}
