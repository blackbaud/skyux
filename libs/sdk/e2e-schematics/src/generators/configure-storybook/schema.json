{
  "$schema": "http://json-schema.org/schema",
  "$id": "configure-storybook",
  "title": "Configure Storybook",
  "description": "Adds configuration to Storybook.",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the project using Storybook.",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    }
  },
  "additionalProperties": false
}
