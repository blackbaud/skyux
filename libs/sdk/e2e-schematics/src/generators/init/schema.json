{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Init",
  "title": "Configure E2E environment"
}
