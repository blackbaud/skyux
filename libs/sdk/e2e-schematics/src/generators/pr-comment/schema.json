{
  "$schema": "http://json-schema.org/schema",
  "$id": "PrComment",
  "title": "PR Comment",
  "description": "Generate markdown for a PR comment or a PR preview README.md",
  "type": "object",
  "properties": {
    "pr": {
      "type": "number",
      "description": "PR Number",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "Which PR is this for?"
    },
    "storybooks": {
      "type": "string",
      "description": "Comma separated list of storybooks to include in the comment",
      "alias": "s",
      "x-prompt": "Which storybook projects should be listed?"
    },
    "apps": {
      "type": "string",
      "description": "Comma separated list of apps to include in the comment",
      "alias": "a"
    }
  },
  "required": ["pr", "storybooks"]
}
