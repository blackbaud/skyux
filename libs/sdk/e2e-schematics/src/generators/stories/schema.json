{
  "$schema": "http://json-schema.org/schema",
  "$id": "Stories",
  "title": "Create Storybook stories/specs",
  "description": "Creates Storybook stories/specs for all Angular components declared in a project.",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What's the name of the project for which you want to generate stories?",
      "aliases": [
        "app",
        "storybook-app",
        "storybookapp",
        "storybookApp",
        "storybook"
      ]
    },
    "generateCypressSpecs": {
      "type": "boolean",
      "description": "Specifies whether to automatically generate `*.spec.ts` files in the Cypress e2e app generated by the `cypress-configure` generator.",
      "default": true
    },
    "cypressProject": {
      "type": "string",
      "description": "The Cypress project to generate the stories under. This is inferred from `name` by default."
    }
  },
  "additionalProperties": false,
  "required": ["project"]
}
