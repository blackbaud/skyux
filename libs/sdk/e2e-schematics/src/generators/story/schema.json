{
  "$schema": "http://json-schema.org/schema",
  "$id": "Story",
  "title": "Create a new story with a new component and corresponding tests",
  "description": "Create a new story with a new component and corresponding tests",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the new storybook component?"
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "x-prompt": "What's the name of the storybook application for which you want to generate the story?",
      "aliases": [
        "app",
        "storybook-app",
        "storybookapp",
        "storybookApp",
        "storybook"
      ]
    },
    "generateCypressSpecs": {
      "type": "boolean",
      "description": "Specifies whether to automatically generate `*.spec.ts` files in the Cypress e2e app generated by the `cypress-configure` generator.",
      "default": true
    },
    "cypressProject": {
      "type": "string",
      "description": "The Cypress project to generate the stories under. This is inferred from `project` by default."
    },
    "includeTests": {
      "type": "boolean",
      "description": "Include tests in the library.",
      "default": false
    }
  },
  "additionalProperties": false,
  "required": ["name", "project"]
}
