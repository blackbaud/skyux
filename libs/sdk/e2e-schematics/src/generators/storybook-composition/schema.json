{
  "$schema": "http://json-schema.org/schema",
  "$id": "storybook-composition",
  "type": "object",
  "properties": {
    "projectsJson": {
      "type": "string",
      "description": "JSON array of projects to include"
    },
    "baseUrl": {
      "type": "string",
      "description": "Web address containing published Storybooks."
    }
  },
  "required": ["projectsJson", "baseUrl"]
}
