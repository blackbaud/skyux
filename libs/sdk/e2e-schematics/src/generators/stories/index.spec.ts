import {
  applicationGenerator,
  componentGenerator,
  storybookConfigurationGenerator,
} from '@nrwl/angular/generators';
import { Tree, readProjectConfiguration } from '@nrwl/devkit';
import { createTreeWithEmptyV1Workspace } from '@nrwl/devkit/testing';
import { Linter } from '@nrwl/linter';

import storiesGenerator from './index';
import { StoriesGeneratorSchema } from './schema';

describe('stories generator', () => {
  let appTree: Tree;
  let options: StoriesGeneratorSchema;

  beforeEach(() => {
    appTree = createTreeWithEmptyV1Workspace();
    options = {
      project: 'test',
      generateCypressSpecs: true,
    };
  });

  it('should run successfully', async () => {
    await applicationGenerator(appTree, {
      name: 'test',
    });
    await storybookConfigurationGenerator(appTree, {
      name: 'test',
      generateCypressSpecs: false,
      configureCypress: true,
      linter: Linter.None,
      generateStories: false,
    });
    await storiesGenerator(appTree, options);
    const config = readProjectConfiguration(appTree, 'test');
    expect(config).toBeDefined();
    expect(
      appTree.read('apps/test/src/app/nx-welcome.component.stories.ts', 'utf-8')
    ).toMatchSnapshot();
  });

  it('should error if a project name is not given', async () => {
    await applicationGenerator(appTree, {
      name: 'test',
    });
    await storybookConfigurationGenerator(appTree, {
      name: 'test',
      generateCypressSpecs: false,
      configureCypress: true,
      linter: Linter.None,
      generateStories: false,
    });
    options.project = undefined;
    try {
      await storiesGenerator(appTree, options);
    } catch (e) {
      if (!(e instanceof Error)) {
        fail('should have thrown error');
      }
      expect(e.message).toBe('Project name not specified');
    }
  });

  it('should generate folder path', async () => {
    await applicationGenerator(appTree, {
      name: 'test',
    });
    await componentGenerator(appTree, {
      name: 'path/to/nested',
      project: 'test',
    });
    await storybookConfigurationGenerator(appTree, {
      name: 'test',
      generateCypressSpecs: false,
      configureCypress: true,
      linter: Linter.None,
      generateStories: false,
    });
    await storiesGenerator(appTree, options);
    const config = readProjectConfiguration(appTree, 'test');
    expect(config).toBeDefined();
    expect(
      appTree.read(
        'apps/test/src/app/path/to/nested/nested.component.stories.ts',
        'utf-8'
      )
    ).toMatchSnapshot();
    expect(
      appTree.read(
        'apps/test-e2e/src/integration/path/to/nested/nested.component.spec.ts',
        'utf-8'
      )
    ).toMatchSnapshot();
  });

  it('should use -storybook project', async () => {
    await applicationGenerator(appTree, {
      name: 'test',
    });
    await applicationGenerator(appTree, {
      name: 'test-storybook',
    });
    await storybookConfigurationGenerator(appTree, {
      name: 'test-storybook',
      generateCypressSpecs: false,
      configureCypress: true,
      linter: Linter.None,
      generateStories: false,
    });
    await storiesGenerator(appTree, options);
    const config = readProjectConfiguration(appTree, 'test-storybook');
    expect(config).toBeDefined();
    expect(
      appTree.read(
        'apps/test-storybook/src/app/nx-welcome.component.stories.ts',
        'utf-8'
      )
    ).toMatchSnapshot();
    expect(
      appTree.read(
        'apps/test-storybook-e2e/src/integration/nx-welcome.component.spec.ts',
        'utf-8'
      )
    ).toMatchSnapshot();
  });

  it('should throw an error when the project does not exist', async () => {
    try {
      await storiesGenerator(appTree, options);
      fail();
    } catch (e) {
      expect((e as Error).message).toEqual('Unable to find project test');
    }
  });

  it('should throw an error when the project does not have storybook', async () => {
    await applicationGenerator(appTree, {
      name: 'test',
    });
    try {
      await storiesGenerator(appTree, options);
      fail();
    } catch (e) {
      expect((e as Error).message).toEqual(
        'Unable to find project test-storybook'
      );
    }
  });

  it('should throw an error when the project does not have storybook configured properly', async () => {
    await applicationGenerator(appTree, {
      name: 'test',
    });
    await applicationGenerator(appTree, {
      name: 'test-storybook',
    });
    try {
      await storiesGenerator(appTree, options);
      fail();
    } catch (e) {
      expect((e as Error).message).toEqual(
        'Storybook is not configured for test-storybook'
      );
    }
  });
});
