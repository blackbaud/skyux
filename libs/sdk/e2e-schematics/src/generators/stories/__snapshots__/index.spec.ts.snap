// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories generator should generate folder path 1`] = `
"import { AppModule } from '../../../app.module';
import { NestedComponent } from './nested.component';
import { Meta, moduleMetadata } from '@storybook/angular';

export default {
  id: 'nestedcomponent-nested',
  title: 'Components/To/Nested',
  component: NestedComponent,
  decorators: [
    moduleMetadata({
      imports: [AppModule],
    }),
  ],
} as Meta<NestedComponent>;
export const Nested = {
  render: (args: NestedComponent) => ({
    props: args,
  }),
  args: {},
};
"
`;

exports[`stories generator should generate folder path 2`] = `
"import { E2eVariations } from '@skyux-sdk/e2e-schematics';

describe('test', () => {
  E2eVariations.forEachTheme((theme) => {
    describe(\`in \${theme} theme\`, () => {
      beforeEach(() =>
        cy.visit(
          \`/iframe.html?globals=theme:\${theme}&id=nestedcomponent-nested--nested\`
        )
      );
      it('should render the component', () => {
        cy.get('proj-nested')
          .should('exist')
          .should('be.visible')
          .screenshot(\`nestedcomponent-nested--nested-\${theme}\`)
          .percySnapshot(\`nestedcomponent-nested--nested-\${theme}\`, {
            widths: E2eVariations.DISPLAY_WIDTHS,
          });
      });
    });
  });
});
"
`;

exports[`stories generator should run successfully 1`] = `
"import { AppModule } from './app.module';
import { NxWelcomeComponent } from './nx-welcome.component';
import { Meta, moduleMetadata } from '@storybook/angular';

export default {
  id: 'nxwelcomecomponent-nxwelcome',
  title: 'Components/Nx Welcome',
  component: NxWelcomeComponent,
  decorators: [
    moduleMetadata({
      imports: [AppModule],
    }),
  ],
} as Meta<NxWelcomeComponent>;
export const NxWelcome = {
  render: (args: NxWelcomeComponent) => ({
    props: args,
  }),
  args: {},
};
"
`;

exports[`stories generator should use -storybook project 1`] = `
"import { AppModule } from './app.module';
import { NxWelcomeComponent } from './nx-welcome.component';
import { Meta, moduleMetadata } from '@storybook/angular';

export default {
  id: 'nxwelcomecomponent-nxwelcome',
  title: 'Components/Nx Welcome',
  component: NxWelcomeComponent,
  decorators: [
    moduleMetadata({
      imports: [AppModule],
    }),
  ],
} as Meta<NxWelcomeComponent>;
export const NxWelcome = {
  render: (args: NxWelcomeComponent) => ({
    props: args,
  }),
  args: {},
};
"
`;

exports[`stories generator should use -storybook project 2`] = `
"import { E2eVariations } from '@skyux-sdk/e2e-schematics';

describe('test-storybook', () => {
  E2eVariations.forEachTheme((theme) => {
    describe(\`in \${theme} theme\`, () => {
      beforeEach(() =>
        cy.visit(
          \`/iframe.html?globals=theme:\${theme}&id=nxwelcomecomponent-nxwelcome--nx-welcome\`
        )
      );
      it('should render the component', () => {
        cy.get('proj-nx-welcome')
          .should('exist')
          .should('be.visible')
          .screenshot(\`nxwelcomecomponent-nxwelcome--nx-welcome-\${theme}\`)
          .percySnapshot(\`nxwelcomecomponent-nxwelcome--nx-welcome-\${theme}\`, {
            widths: E2eVariations.DISPLAY_WIDTHS,
          });
      });
    });
  });
});
"
`;
